### PR TITLE
Generalize example generator and isolate examples

### PR DESCRIPTION
**Pull request by planger**
#### What it does
* Centralize definition of examples
* Generalize example generator
* Introduce contribution point for examples
* Support merging of new launch and task configs with existing ones
* Add reveal navigator after project generation

One disadvantage of this change is that we can't easily specify dependent tasks by name as the name now may depend on the specified folder name. We could make the tasks.json and launch.json generation more clever but it feels like a pretty heavy solution for such a minor added feature. What do you think @sdirix?

Contributed on behalf of STMicroelectronics

#### How to test
* Test creating CDT Blueprint Example projects with or without a specified folder.
* Test that the generated launches and tasks are correct and working.

### CODE REVIEW

**Comment by planger** [comment]
@sdirix Did you have a chance to look at the PR above, especially with respect to making the launch.json and tasks.json generation more clever to allow for references between them after we change the name automatically, in case there are more than one projects?

File: theia-extensions/blueprint-examples/src/node/blueprint-examples-contribution.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        return [{
+            id: CdtCloudBlueprintExamples.CMAKE_EXAMPLE,
+            label: 'CMake example',
+            welcomeFile: 'CMAKE_EXAMPLE_README.md',
+            resourcesPath: new URI(module.path).resolve('../../resources/cmake-example').normalizePath().toString(),
+            launches: [{
+                'type': 'gdb',
+                'request': 'launch',
+                'name': 'Debug Example C++',
+                'program': '${targetFolder}/Example',
```
Comment (sdirix) [inline theia-extensions/blueprint-examples/src/node/blueprint-examples-contribution.ts:35]: This `${targetFolder}` replacement is kind of magic and is done via regex later on. Would it make sense to transform the examples definition here to a parameterized function instead?

For example:
```
return [
    {
            id: CdtCloudBlueprintExamples.CMAKE_EXAMPLE,
            label: 'CMake example',
            welcomeFile: 'CMAKE_EXAMPLE_README.md',
            resourcesPath: new URI(module.path).resolve('../../resources/cmake-example').normalizePath().toString(),
            configurations: (options) => {
                launches: [
                    {
                        'type': 'gdb',
                        'request': 'launch',
                        'name': 'Debug Example C++',
                        'program': `${options.targetFolder}/Example`,
                        'initCommands': ['tbreak main']
                    }
                ]
            }
            // etc.
    }
]
```

**Review by sdirix — COMMENTED** [review]
Works for me! I have some minor comments

File: theia-extensions/blueprint-examples/src/node/blueprint-examples-contribution.ts (hunk 1 above)
Comment (planger) [reply theia-extensions/blueprint-examples/src/node/blueprint-examples-contribution.ts]: Great idea!

**Comment by planger** [comment]
Thanks for the great review! I've pushed 4e60919 to address your feedback! Please let me know what you think!

**Pull request merged**